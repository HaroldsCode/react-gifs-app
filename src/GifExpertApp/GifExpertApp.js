import { useState } from "react";
import { AddCategory } from './Components/AddCategory';
import { GifGrid } from "./Containers/GifGrid";

export const GifExpertApp = ({categoryValue}) => {
    const [categories, setCategories] = useState(categoryValue);
    return (
        <>
            <div className="introduce">
                <h1 className="introduce__title">GifExpertApp</h1>
                <p className="introduce__copy">Bienvenido, en este sitio web encontraras gifs divertidos para pasar el rato.</p>
                <AddCategory setNumbers={setCategories}/>
            </div>
            {(categories !== '') && (
                <div className="group">
                    <GifGrid key={categories} category={categories}/>
                </div>
            )}
        </>
    )
}

GifExpertApp.defaultProps = {
    categoryValue: '',
}