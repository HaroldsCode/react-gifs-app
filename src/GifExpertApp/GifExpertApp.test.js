import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from './GifExpertApp';

describe('GitExpertApp tests', () => {
    test('should display the component correctly', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should display the category', () => {
        const category = 'one punch';
        const wrapper = shallow(<GifExpertApp categoryValue={category}/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(1)
    })
    
})
