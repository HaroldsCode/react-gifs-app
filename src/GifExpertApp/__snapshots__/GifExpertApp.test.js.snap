// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitExpertApp tests should display the category 1`] = `
<Fragment>
  <div
    className="introduce"
  >
    <h1
      className="introduce__title"
    >
      GifExpertApp
    </h1>
    <p
      className="introduce__copy"
    >
      Bienvenido, en este sitio web encontraras gifs divertidos para pasar el rato.
    </p>
    <AddCategory
      setNumbers={[Function]}
    />
  </div>
  <div
    className="group"
  >
    <GifGrid
      category="one punch"
      key="one punch"
    />
  </div>
</Fragment>
`;

exports[`GitExpertApp tests should display the component correctly 1`] = `
<Fragment>
  <div
    className="introduce"
  >
    <h1
      className="introduce__title"
    >
      GifExpertApp
    </h1>
    <p
      className="introduce__copy"
    >
      Bienvenido, en este sitio web encontraras gifs divertidos para pasar el rato.
    </p>
    <AddCategory
      setNumbers={[Function]}
    />
  </div>
</Fragment>
`;
